### PR TITLE
Schematic Symbols

### PR DESCRIPTION
**Pull request by evanhekman**
# Pull Request

Go to the `Preview` tab and select the appropriate template for your pull request:

- [Issue](?expand=1&template=issue.md)
- [RC](?expand=1&template=rc.md)

Adds flowmeters, ball valves, compressors, heat exchangers, flame arrestors and updates agitators.

### CODE REVIEW

File: pluto/src/vis/schematic/registry.ts [hunk 1]
```diff
@@ -1020,18 +1633,34 @@ export const SYMBOLS: Record<Variant, Spec<any>> = {
...
+  springLoadedReliefValve,
+  angledSpringLoadedReliefValve,
   pump,
   pistonPump,
   screwPump,
   cavityPump,
-  vacuumPump,
+  diaphragmPump,
+  ejectionPump,
+  // vacuumPump,
```
Comment (pjdotson) [inline pluto/src/vis/schematic/registry.ts:1657]: uncomment

File: pluto/src/vis/schematic/registry.ts [hunk 2]
```diff
@@ -730,19 +922,19 @@ const switch_: Spec<SwitchProps> = {
...
+//   Symbol: VacuumPump,
+//   Form: CommonToggleForm,
+//   defaultProps: (t) => ({
+//     color: t.colors.gray.l9.rgba255,
+//     ...zeroLabel("Vacuum Pump"),
+//     ...ZERO_TOGGLE_PROPS,
+//   }),
+//   Preview: Primitives.VacuumPump,
+//   zIndex: Z_INDEX_UPPER,
+// };
```
Comment (pjdotson) [inline pluto/src/vis/schematic/registry.ts:937]: uncomment

File: pluto/src/vis/schematic/registry.ts [hunk 3]
```diff
@@ -199,11 +288,30 @@ const VARIANTS = [
   "tank",
   "textBox",
   "threeWayValve",
-  "vacuumPump",
+  // "vacuumPump",
```
Comment (pjdotson) [inline pluto/src/vis/schematic/registry.ts:291]: uncomment

File: pluto/src/vis/schematic/registry.ts [hunk 4]
```diff
@@ -130,15 +184,33 @@ import {
   type ThreeWayValveProps,
   TJunction,
   type TJunctionProps,
-  VacuumPump,
-  type VacuumPumpProps,
+  CrossJunction,
+  type CrossJunctionProps,
+  // VacuumPump,
+  // type VacuumPumpProps,
```
Comment (pjdotson) [inline pluto/src/vis/schematic/registry.ts:190]: uncomment

File: pluto/src/vis/schematic/primitives/Primitives.tsx [hunk 5]
```diff
@@ -35,6 +35,7 @@ import { CSS } from "@/css";
 import { Input } from "@/input";
 import { Text } from "@/text";
 import { Theming } from "@/theming";
+import { isDisabled } from "@testing-library/user-event/dist/cjs/utils/index.js";
```
Comment (pjdotson) [inline pluto/src/vis/schematic/primitives/Primitives.tsx:38]: remove

File: pluto/src/vis/schematic/primitives/Primitives.tsx
Comment (pjdotson) [inline pluto/src/vis/schematic/primitives/Primitives.tsx:1]: Would be good to use prettier to format this file with 88 line length limit

File: pluto/src/vis/schematic/primitives/Primitives.tsx [hunk 6]
```diff
@@ -1935,25 +2201,16 @@ export const Agitator = ({
...
   <Toggle {...props} className={CSS(CSS.B("agitator"))}>
     <HandleBoundary orientation={orientation}>
-      <Handle
-        location="top"
-        orientation={orientation}
-        left={50}
-        top={100 / height}
-        id="1"
-      />
+      <Handle location="top" orientation={orientation} left={50} top={2.27} id="4"/>
```
Comment (pjdotson) [inline pluto/src/vis/schematic/primitives/Primitives.tsx:2204]: Why was this changed?

File: pluto/src/vis/schematic/primitives/Primitives.tsx [hunk 7]
```diff
@@ -1969,22 +2226,16 @@ export const PropellerAgitator = ({
...
   <Toggle {...props} className={CSS(CSS.B("agitator"))}>
     <HandleBoundary orientation={orientation}>
-      <Handle
-        location="top"
-        orientation={orientation}
-        left={50.5814}
-        top={200 / height}
-        id="4"
-      />
+      <Handle location="top" orientation={orientation} left={50} top={2.27} id="4"/>
```
Comment (pjdotson) [inline pluto/src/vis/schematic/primitives/Primitives.tsx:2229]: why changed?

File: pluto/src/vis/schematic/primitives/Primitives.tsx [hunk 8]
```diff
@@ -2000,22 +2251,16 @@ export const FlatBladeAgitator = ({
...
   <Toggle {...props} className={CSS(CSS.B("agitator"))}>
     <HandleBoundary orientation={orientation}>
-      <Handle
-        location="top"
-        orientation={orientation}
-        left={50}
-        top={100 / height}
-        id="4"
-      />
+      <Handle location="top" orientation={orientation} left={50} top={2.27} id="4"/>
```
Comment (pjdotson) [inline pluto/src/vis/schematic/primitives/Primitives.tsx:2254]: why changed?

File: pluto/src/vis/schematic/primitives/Primitives.tsx [hunk 9]
```diff
@@ -2032,47 +2277,18 @@ export const PaddleAgitator = ({
...
```
Comment (pjdotson) [inline pluto/src/vis/schematic/primitives/Primitives.tsx:2280]: why changed?

File: pluto/src/vis/schematic/primitives/Primitives.tsx [hunk 10]
```diff
@@ -2089,25 +2305,17 @@ export const CrossBeamAgitator = ({
...
```
Comment (pjdotson) [inline pluto/src/vis/schematic/primitives/Primitives.tsx:2308]: why changed?

File: pluto/src/vis/schematic/primitives/Primitives.tsx [hunk 11]
```diff
@@ -2124,25 +2332,16 @@ export const HelicalAgitator = ({
...
   <Toggle {...props} className={CSS(CSS.B("agitator"))}>
     <HandleBoundary orientation={orientation}>
-      <Handle
-        location="top"
-        left={50}
-        top={100 / height}
-        id="4"
-        orientation={orientation}
-      />
+      <Handle location="top" orientation={orientation} left={50} top={2.27} id="4"/>
```
Comment (pjdotson) [inline pluto/src/vis/schematic/primitives/Primitives.tsx:2335]: why changed?

File: pluto/src/vis/schematic/primitives/Primitives.tsx [hunk 12]
```diff
@@ -2162,19 +2361,18 @@ export const Compressor = ({
     orientation={orientation}
   >
     <HandleBoundary orientation={orientation}>
-      <Handle location="left" orientation={orientation} left={1.5} top={41} id="1" />
-      <Handle location="right" orientation={orientation} left={95} top={3} id="2" />
+      <Handle location="left" orientation={orientation} left={4.55} top={50} id="1" />
```
Comment (pjdotson) [inline pluto/src/vis/schematic/primitives/Primitives.tsx:2364]: why was this changed?

**Review by pjdotson — CHANGES_REQUESTED** [review]
New symbols look really good and clean. A couple (relatively minor) issues:
- wanted to know why some existing symbols like the agitators and the compressor were changed
- vacuum pump got commented out in `registry.ts`
- These files should be formatted with [Prettier](https://prettier.io), there is a VSCode extension that you can use to do this automatically if you put `formatOnSave` in your VSCode settings to true

File: pluto/src/vis/schematic/registry.ts (hunk 1 above)
Comment (evanhekman) [reply pluto/src/vis/schematic/registry.ts:1657]: Vacuum Pump was commented out because it was replaced by Compressor, could be changed back but we should remove Compressor so there aren't duplicates under different names.

**Comment by pjdotson** [comment]
@evanhekman Can you backmarge the main `synnaxlabs:rc` branch into `evanhekman:main`?

File: pluto/src/vis/schematic/registry.ts (hunk 1 above)
Comment (pjdotson) [reply pluto/src/vis/schematic/registry.ts:1657]: We need to keep vacuum pump because we have customers using it, could you instead remove the compressor? If you want, you can change the name and default label of the vacuum pump to be compressor, but the key in `registry.ts` and the `aetherKey` in `Symbols.tsx` needs to remain the same as the existing vacuum pump key.

**Comment by emilbon99** [comment]
@pjdotson There's a commit that says it was backmerged

**Comment by pjdotson** [comment]
@evanhekman sorry just checked the old source code again - I realized vacuum pump and compressor both already existed. Can you make sure that you don't delete either of them?

**Review by pjdotson — APPROVED** [review]

**Review by emilbon99 — APPROVED** [review]
